### PR TITLE
Added circle_sort.php

### LINKED ISSUE #2129: Circle Sort
- [x] C
- [x] C++
- [ ] C#
- [ ] CoffeeScript
- [ ] Dart
- [ ] Go - @aakashdinkar
- [x] Java
- [ ] JS - @HariniJeyaraman
- [x] Kotlin - @myselfanuj
- [x] Php - @mharshita
- [x] Python
- [x] Readme
- [x] Ruby - @HardevKhandhar
- [ ] Rust
- [ ] TypeScript

### PR DESCRIPTION
**Pull request by mharshita**
Fixes #2129 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Added circle_sort.php

#### Languages Used:
- PHP

#### Files Added:
- circle_sort.php

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by mharshita** [comment]
@jainaman224  @somya-kapoor  Please review my PR

File: Circle_Sort/circle_sort.php [hunk 1]
```diff
@@ -0,0 +1,96 @@
+<?php
```
Comment (somya-kapoor) [inline Circle_Sort/circle_sort.php:1]: Change the file name to Circle_Sort/Circle_Sort.php

File: Circle_Sort/circle_sort.php [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+//Circle sort algorithm can be visualized by drawing concentric circles on an array of integers. 
+//The elements of the array lying on the same circle diametrically opposite to each other are compared and if found in the wrong order they are swapped.
```
Comment (somya-kapoor) [inline Circle_Sort/circle_sort.php:4]: Divide the sentence, in a single line there should be around 15 -16 words only

File: Circle_Sort/circle_sort.php [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+//function to perform circular swaps recursively.
+//this function returns true if there was a swap.
+//operation performed.
+
+function circleSortRec($a, $low, $high)
+{
+    $swapped = false;
+    
+    //base case
+
```
Comment (somya-kapoor) [inline Circle_Sort/circle_sort.php:16]: Remove this line

File: Circle_Sort/circle_sort.php [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    while(circleSortRec($a , 0 , $n-1))
+    {
+        ;
+    }
+}
+
+//Driver Program
+
+$a = array(7, 5, 3, 1, 2, 4, 6, 8);
```
Comment (somya-kapoor) [inline Circle_Sort/circle_sort.php:75]: Take dynamic input

File: Circle_Sort/circle_sort.php [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+
+echo "\n Sorted: ";
+
+for($i = 0; $i < $n; $i++)
+{
+    echo "$a[$i] ";
+}
+
+return 0;
+
```
Comment (somya-kapoor) [inline Circle_Sort/circle_sort.php:95]: Add sample input/output

File: Circle_Sort/circle_sort.php [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+    
+    //base case
+
+    if($low == $high)
+    {
+        return false;
+    }
+
+    //storing upper and lower bounds of list to be used in the recursive case.
+
```
Comment (somya-kapoor) [inline Circle_Sort/circle_sort.php:23]: Remove this line

File: Circle_Sort/circle_sort.php [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+    $lo = $low;
+    $hi = $high;
+
+    while($lo < $hi)
+    {
+        //swaps the pair of elements if True.
+
```
Comment (somya-kapoor) [inline Circle_Sort/circle_sort.php:30]: Remove this line

File: Circle_Sort/circle_sort.php [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+            $a[$lo] = $a[$hi];
+            $a[$hi] = $temp;
+            $swapped = true;
+        }
+        $lo++;
+        $hi--;
+    }
+
+    //special case arises only for list of odd size.
+
```
Comment (somya-kapoor) [inline Circle_Sort/circle_sort.php:43]: Remove this line

File: Circle_Sort/circle_sort.php [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+    if($a[$lo] > $a[$hi+1])
+    {
+        $t = $a[$low];
+        $a[$low] = $a[$hi+1];
+        $a[$hi+1] = $t;
+        $swapped = true;
+    }
+
+    //recursive case to check the traverse lists as sub lists.
+
```
Comment (somya-kapoor) [inline Circle_Sort/circle_sort.php:54]: Remove this line

File: Circle_Sort/circle_sort.php [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+    $mid = ($high - $low)/2;
+    $firstHalf = circleSortRec($a , $low , $low+$mid);
+    $secondHalf = circleSortRec($a , $low+$mid+1 , $high);
+    return $swapped || $firstHalf || $secondHalf;
+}
+
+//this function mainly calls circleSortRec
+
```
Comment (somya-kapoor) [inline Circle_Sort/circle_sort.php:62]: Remove this line

File: Circle_Sort/circle_sort.php [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+function circleSort($a , $n)
+{
+    //keep calling circleSortRec while there is a swap operation.
+
```
Comment (somya-kapoor) [inline Circle_Sort/circle_sort.php:66]: Remove this line

**Comment by somya-kapoor** [comment]
Remove the 2nd file.

**Comment by mharshita** [comment]
@jainaman224  @somya-kapoor  The required changes have been done. Please review.

File: Circle_Sort/circle_sort.php (hunk 1 above)
Comment (mharshita) [reply Circle_Sort/circle_sort.php]: @somya-kapoor  file name changed.

File: Circle_Sort/circle_sort.php (hunk 2 above)
Comment (mharshita) [reply Circle_Sort/circle_sort.php]: @somya-kapoor done the requied changes.

File: Circle_Sort/circle_sort.php (hunk 3 above)
Comment (mharshita) [reply Circle_Sort/circle_sort.php]: @somya-kapoor  done the required changes

File: Circle_Sort/circle_sort.php (hunk 4 above)
Comment (mharshita) [reply Circle_Sort/circle_sort.php]: @somya-kapoor  done the required changes.

File: Circle_Sort/circle_sort.php (hunk 5 above)
Comment (mharshita) [reply Circle_Sort/circle_sort.php]: @somya-kapoor  sample input and output are added.

File: Circle_Sort/circle_sort.php (hunk 6 above)
Comment (mharshita) [reply Circle_Sort/circle_sort.php]: @somya-kapoor  done the required changes.

File: Circle_Sort/circle_sort.php (hunk 7 above)
Comment (mharshita) [reply Circle_Sort/circle_sort.php]: @somya-kapoor  done the required changes.

File: Circle_Sort/circle_sort.php (hunk 8 above)
Comment (mharshita) [reply Circle_Sort/circle_sort.php]: @somya-kapoor  done the required changes.

File: Circle_Sort/circle_sort.php (hunk 9 above)
Comment (mharshita) [reply Circle_Sort/circle_sort.php]: @somya-kapoor  done the required changes.

File: Circle_Sort/circle_sort.php (hunk 10 above)
Comment (mharshita) [reply Circle_Sort/circle_sort.php]: @somya-kapoor  done the required changes.

File: Circle_Sort/circle_sort.php (hunk 11 above)
Comment (mharshita) [reply Circle_Sort/circle_sort.php]: @somya-kapoor  done the required changes.

**Comment by mharshita** [comment]
@jainaman224  @asha15  @somya-kapoor  Please review my PR. The required changes have been done.

File: Circle_Sort/Circle_Sort.php [hunk 12]
```diff
@@ -0,0 +1,85 @@
...
+            $a[$lo] = $a[$hi];
+            $a[$hi] = $temp;
+            $swapped = true;
+        }
+        $lo++;
+        $hi--;
+    }
+      
+    if($lo == $hi)
+    if($a[$lo] > $a[$hi+1])
```
Comment (somya-kapoor) [inline Circle_Sort/Circle_Sort.php:32]: Add space around +

File: Circle_Sort/Circle_Sort.php [hunk 13]
```diff
@@ -0,0 +1,85 @@
...
+    {
+        $t = $a[$low];
+        $a[$low] = $a[$hi+1];
```
Comment (somya-kapoor) [inline Circle_Sort/Circle_Sort.php:35]: Add space around +

File: Circle_Sort/Circle_Sort.php [hunk 14]
```diff
@@ -0,0 +1,85 @@
...
+        $a[$hi+1] = $t;
```
Comment (somya-kapoor) [inline Circle_Sort/Circle_Sort.php:36]: Add space around +

File: Circle_Sort/Circle_Sort.php [hunk 15]
```diff
@@ -0,0 +1,85 @@
...
+        $swapped = true;
+    }
+
+    $mid = ($high - $low)/2;
+    $firstHalf = circleSortRec($a , $low , $low+$mid);
```
Comment (somya-kapoor) [inline Circle_Sort/Circle_Sort.php:41]: Add space around +

File: Circle_Sort/Circle_Sort.php [hunk 16]
```diff
@@ -0,0 +1,85 @@
...
+    $secondHalf = circleSortRec($a , $low+$mid+1 , $high);
```
Comment (somya-kapoor) [inline Circle_Sort/Circle_Sort.php:42]: Add space around +

File: Circle_Sort/Circle_Sort.php [hunk 17]
```diff
@@ -0,0 +1,85 @@
...
+    return $swapped || $firstHalf || $secondHalf;
+}
+
+function circleSort($a , $n)
+{
+
+    while(circleSortRec($a , 0 , $n-1))
```
Comment (somya-kapoor) [inline Circle_Sort/Circle_Sort.php:49]: Add space around -

File: Circle_Sort/Circle_Sort.php [hunk 18]
```diff
@@ -0,0 +1,85 @@
...
+    {
+        ;
+    }
+}
+
+$a = explode(' ',readline());
+$n = sizeof($a)/sizeof($a[0]);
```
Comment (somya-kapoor) [inline Circle_Sort/Circle_Sort.php:56]: Add space around /

File: Circle_Sort/Circle_Sort.php [hunk 19]
```diff
@@ -0,0 +1,85 @@
...
+    echo "$a[$i] ";
+}
+
+return 0;
+
+/*
+Input:
+7 5 3 1 2 4 6 8
+
+ Output:
```
Comment (somya-kapoor) [inline Circle_Sort/Circle_Sort.php:80]: Remove space before output

**Comment by somya-kapoor** [comment]
After making the changes, please squash the commits.

File: Circle_Sort/Circle_Sort.php (hunk 12 above)
Comment (mharshita) [reply Circle_Sort/Circle_Sort.php]: @somya-kapoor  Done the required changes.

File: Circle_Sort/Circle_Sort.php (hunk 13 above)
Comment (mharshita) [reply Circle_Sort/Circle_Sort.php]: @somya-kapoor Done the required changes.

File: Circle_Sort/Circle_Sort.php (hunk 14 above)
Comment (mharshita) [reply Circle_Sort/Circle_Sort.php]: @somya-kapoor  Done the required changes.

File: Circle_Sort/Circle_Sort.php (hunk 15 above)
Comment (mharshita) [reply Circle_Sort/Circle_Sort.php]: @somya-kapoor  Done the required changes.

File: Circle_Sort/Circle_Sort.php (hunk 16 above)
Comment (mharshita) [reply Circle_Sort/Circle_Sort.php]: @somya-kapoor  Done the required changes.

File: Circle_Sort/Circle_Sort.php (hunk 17 above)
Comment (mharshita) [reply Circle_Sort/Circle_Sort.php]: @somya-kapoor Done the required changes.

File: Circle_Sort/Circle_Sort.php (hunk 18 above)
Comment (mharshita) [reply Circle_Sort/Circle_Sort.php]: @somya-kapoor  Done the required changes.

File: Circle_Sort/Circle_Sort.php (hunk 19 above)
Comment (mharshita) [reply Circle_Sort/Circle_Sort.php]: @somya-kapoor Done the required changes.

**Comment by mharshita** [comment]
@somya-kapoor  Please review my PR. The required changes have been done and commits have also been squashed.

File: Circle_Sort/Circle_Sort.php [hunk 20]
```diff
@@ -0,0 +1,85 @@
...
+    $mid = ($high - $low) / 2;
+    $firstHalf = circleSortRec($a , $low , $low + $mid);
+    $secondHalf = circleSortRec($a , $low + $mid + 1 , $high);
+    return $swapped || $firstHalf || $secondHalf;
+}
+
+function circleSort($a , $n)
+{
+
+    while(circleSortRec($a , 0 , $n - 1))
```
Comment (singh-shreya6) [inline Circle_Sort/Circle_Sort.php:49]: Can't this be replaced by while(circleSortRec($a , 0 , $n - 1));

Comment (mharshita) [reply Circle_Sort/Circle_Sort.php]: @singh-shreya6  done the required changes.

**Comment by mharshita** [comment]
@singh-shreya6  I have made the required changes and squashed the commits. Please review.

**Pull request merged**